app/src/main/java/in/testpress/testpress/authenticator/LoginActivity.java

```diff
@@ -484,6 +491,10 @@ protected void finishLogin() {
         sharedPreferences.edit().putBoolean(GCMPreference.SENT_TOKEN_TO_SERVER, false).apply();
         CommonUtils.registerDevice(this, testpressService);
         final Account account = new Account(username, APPLICATION_ID);
+        if (!isPreviouslyLoggedInUser()) {
+            deleteDownloadedVideos();
+        }
+        saveUsernameToPreference();
```

Comment:
This can be `saveLoggedInUsername` because preference is an internal detail. It need to be shared preference, you can use db or any other method.